### ISSUE TITLE
Duplicate extension rules across naming rules silently break checks

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#140]
## Description

When `check.toml` contains two naming rules with the same extension but different case requirements (e.g., one rule says `ts` should be `kebab-case`, another says `ts` should be `snake_case`), both the Naming check AND the Disable Comments check silently fail. They report "passed" even when there are files that should be flagged as violations.

## Steps to Reproduce

1. Create a `check.toml` with duplicate extension rules:
```toml
[code.naming]
enabled = true

[[code.naming.rules]]
extensions = ["ts", "tsx"]
file_case = "kebab-case"
folder_case = "kebab-case"

[[code.naming.rules]]
extensions = ["ts"]  # Duplicate with conflicting case
file_case = "snake_case"
folder_case = "snake_case"
```
2. Run `cm validate config` - it passes validation
3. Run `cm code check` - Naming shows "passed" even with files that violate naming rules
4. Disable Comments also shows "passed" even with disable comments in files

## Expected Behavior

- Configuration validation should reject duplicate extensions across rules, OR
- The tool should use a deterministic rule (first match, last match, etc.) and still find violations, OR
- At minimum, the tool should not silently fail - it should either work or error

## Actual Behavior

- Configuration validation passes
- Naming and Disable Comments both silently report "passed"
- No violations are found even for files that clearly violate naming conventions

## Impact

HIGH - This could allow naming violations and disable comments to slip through CI checks undetected.

## Version

1.3.0
